src/content/app/genome-browser/hooks/useFocusTrack.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useRef } from 'react';
+import {
+  IncomingActionType,
+  type ReportVisibleTranscriptsAction
+} from '@ensembl/ensembl-genome-browser';
+
+import { useAppSelector, useAppDispatch } from 'src/store';
+import useGenomeBrowserIds from './useGenomeBrowserIds';
+import useGenomeBrowser from 'src/content/app/genome-browser/hooks/useGenomeBrowser';
+
+import { getFocusObjectTrackState } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSelectors';
+
+import { updateObjectTrackStates } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSlice';
+
+import { Status } from 'src/shared/types/status';
+import type { FocusGeneTrack } from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
+import type { FocusObjectIdConstituents } from 'src/shared/types/focus-object/focusObjectTypes';
+
+/**
+ * The purposes of this hook are:
+ * - be responsible for telling genome browser how to display the focus object
+ * - save the focus object to the previously viewed list when it changes
+ *
+ * Only one copy of this hook should be run.
+ */
+
+const useFocusTrack = () => {
+  const genomeBrowserMethods = useGenomeBrowser(); // getting it here once to avoid unnecessarily calling this hook from inside child hooks
+  const {
+    genomeId = '',
+    focusObjectId = '',
+    parsedFocusObjectId = null
+  } = useGenomeBrowserIds();
+  const focusObject = useAppSelector((state) =>
+    getFocusObjectTrackState(state, {
+      genomeId,
+      focusObjectId
+    })
+  );
+
+  useFocusGene({
+    genomeBrowserMethods,
+    focusGene: parsedFocusObjectId?.type === 'gene' ? focusObject : null,
+    focusObjectId,
+    parsedFocusObjectId
+  });
+
+  /**
+   * TODO: move the code for saving previous focus object here
+   */
```

Comment:
Is this for a future task?

Reply:
Yeah, I'll add this in https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1675